### DISCUSSION TITLE
Is the formatting work for async_logger done within the logging thread?

### DISCUSSION THREAD

**Discussion by kof02guy** [gabime/spdlog#2308] [Q&A] [answered]
Hi, @gabime 
I'm on a project which requires an async logger aiming to log my server's operations in a really high qps with little performance impact. And I found spdlog a candidate.
Correct me if I'm wrong. The main purpose of async logger is to offload the work required for logging from the thread triggering the logging. But from the code I saw that std::vformat and details::format_to are both executed in the log_() function, which is actually still in the thread triggering the logging.
Am I right?

**Comment by gabime** [comment, marked as answer]
Right. See #2282 for more info about sync vs async

**Reply by kof02guy** [reply]
Thanks @gabime !
I have another question. We use bthread https://github.com/apache/incubator-brpc/tree/master/src/bthread in out project. So the mutex we use is bthread::mutex, which is a user space mutex, rather than std::mutex. If we use async_logger, the whole thread will be blocked when pushing a message to the queue. 
Do you have any suggestions on this?
